src/main/java/ru/urururu/sanity/cpp/tools/Tool.java

```diff
@@ -1,28 +1,61 @@
 package ru.urururu.sanity.cpp.tools;
 
+import java.io.BufferedReader;
 import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.Collections;
+import java.util.List;
 import java.util.Optional;
 import java.util.Set;
-import java.util.function.Function;
+import java.util.function.BiFunction;
 
 /**
  * @author <a href="mailto:dmitriy.g.matveev@gmail.com">Dmitry Matveev</a>
  */
 public abstract class Tool {
+    final String executable;
+    private final List<String> versionIds;
+
+    Tool(String executable, String version) {
+        this.executable = executable;
+        this.versionIds = evaluateVersionIds(version);
+        System.out.println("version = " + version);
+        System.out.println("versionId = " + versionIds);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS106)